src/bits/bit_field_vec.rs

```diff
@@ -93,20 +93,102 @@ fn mask<W: Word>(bit_width: usize) -> W {
 }
 
 impl<W: Word> BitFieldVec<W, Vec<W>> {
-    /// Create a new zero-initialized vector of given bit width and length.
-    pub fn new(bit_width: usize, len: usize) -> Self {
-        // We need at least one word to handle the case of bit width zero.
+    /// Returns a new BitFieldVec filled with `word`
+    ///
+    /// # Arguments
+    /// * `bit_width` - The bit width of the values stored in the vector.
+    /// * `len` - The length of the vector.
+    /// * `word` - The value to fill the vector with.
+    ///
+    /// # Example
+    /// In the following example, we show how to create a new BitFieldVec
+    /// filled with a given word. In this case, we create a vector with length
+    /// 10 and bit width 4, filled with the value 0b11110000. Do note that the
+    /// number of bits in the word must be equal to the bit width of the vector.
+    ///
+    /// ```rust
+    /// use sux::prelude::*;
+    /// let vec = BitFieldVec::<u8>::fill(4, 10, 0b11110000);
+    ///
+    /// assert_eq!(vec.len(), 10);
+    /// for i in 0..10 {
+    ///     if i % 2 == 0 {
+    ///         assert_eq!(vec.get(i), 0b0000);
+    ///     } else {
+    ///         assert_eq!(vec.get(i), 0b1111);
+    ///     }
+    /// }
+    /// ```
+    pub fn fill(bit_width: usize, len: usize, word: W) -> Self {
+        // we need at least two words to avoid branches in the gets
```

Comment:
The comment contradicts the code.

Reply:
Renamed method to `fill_words` and added method `fill` to create a new vector filled with provided values. I think the method `new_ones` should be renamed as it does not return a vector filled with values equal to one (unless the `bit_width` is equal to one), but returns a vector with all words maxed out.